olp-cpp-sdk-core/tests/thread/TaskContinuationTest.cpp

```diff
@@ -132,37 +112,26 @@ TEST_F(TaskContinuationFixture, CancelAfterRun) {
                    std::function<void(olp::client::HttpResponse)> next) {
             next(olp::client::HttpResponse(200, "OK"));
           })
-          .Finally([&promise,
-                    &result](olp::client::ApiResponse<olp::client::HttpResponse,
-                                                      olp::client::ApiError>
-                                 response) {
-            result = response.GetResult();
-            promise.set_value();
-          });
+          .Finally(
+              [&promise](ResponseType<olp::client::HttpResponse> response) {
+                promise.set_value(response.MoveResult());
+              });
 
   cont.Run();
-  std::this_thread::sleep_for(
-      std::chrono::milliseconds(50));  // simulate some delay
 
   cont.CancelToken().Cancel();
 
-  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(1000)),
-            std::future_status::ready);
-
-  future.get();
+  auto result = future.get();
```

Comment:
Please use the approach described above.

Reply:
done